robotoff/insights/ocr/location.py

```diff
@@ -0,0 +1,243 @@
+            address_end = max(city_end, pc_end) + self.text_extract_distance
+            text_extract = text[max(0, address_start) : min(len(text), address_end)]
+
+            locations.append(
+                {
+                    "country_code": "fr",
+                    "city_name": city.name,
+                    "postal_code": city.postal_code,
+                    "text_extract": text_extract,
+                }
+            )
+
+        return locations
+
+    @staticmethod
+    def get_text(ocr_result: OCRResult) -> str:
+        """Extract text from the OCR result and prepare it.
+
+        Args:
+            ocr_result (OCRResult): The OCR result to process.
+
+        Returns:
+            str: The text extracted and prepared.
+        """
+        text = ocr_result.get_full_text(lowercase=True)
+        if text is None:
+            # Using `OCRResult.text_annotations` directly instead of
+            # `OCRResult.get_text_annotations()` because the latter contains
+            # the text duplicated
+            text = ocr_result.text_annotations[0].text.lower()
+        text = strip_accents_ascii(text)
+        text = text.replace("'", " ").replace("-", " ")
+        return text
+
+    def find_city_names(self, text: str) -> List[Tuple[City, int, int]]:
+        """Find all cities from the search set in the text.
+
+        Args:
+            text (str): Text to search city names in.
+
+        Returns:
+            list of (City, int, int): The list of `City`s which name was found in the
+            text, with the start and end indices of their names locations in the
+            text. Empty list if none found.
+        """
+        return self.cities_processor.extract_keywords(text, span_info=True)
+
+    def find_nearby_postal_code(
+        self, text: str, city: City, city_start: int, city_end: int
+    ) -> Optional[Tuple[str, int, int]]:
+        """Search for a city's postal code close to its name in the text.
+
+        The postal code is searched at a maximum distance of
+        `postal_code_search_distance` from the city name.
+
+        Assumes digit-only postal code, allows non-digit directly next to it. For
+        example, for the city "paris" with postal code "75000", "75000 paris" and
+        "fr75000 paris" will match.
+
+        Args:
+            text (str): The OCR result text.
+            city (City): The `City` for which to search the postal code.
+            city_start (int): Start index of the city name match in `text`.
+            city_end (int): End index of the city name match in `text`.
+
+        Returns:
+            (str, int, int) or None: If the `City`'s postal code was found close to
+            the city name match, it is returned along with its start and end indices
+            in the text. If it was not found, returns None.
+        """
+        if not city.postal_code.isdigit():
+            raise ValueError(f"postal code contains non-digit characters: {city}")
```

Comment:
a logger.error followed by a "return" may be more appropriate, as it would prevent the main process from crashing, in case one of the postal code is malformed.